lua/sunglasses/window.lua

```diff
@@ -132,6 +142,11 @@ function Window:can_shade()
             return "an excluded filetype"
         end
     end
+
+    if self.window_can_shade_callback ~= nil and type(self.window_can_shade_callback) == "function" then
+      return self.window_can_shade_callback(self.window)
```

Comment:
I don't know that I like this but I can't really think of any reasons why aside from "the user shouldn't need to pass a function to determine if they want to shade the window or not".

That is not a good enough reason to exclude this, just figured I would notate my reservation here.